### DISCUSSION TITLE
Development workflow / file structure for Cinnamon Spices applets

### DISCUSSION THREAD

**Discussion by Starcross** [linuxmint/discussions#533] [Development] [answered]
This is a question about how best to work on applets that are already in the `cinnamon-spices-applets` github repo. 

The file structure in the repo follows the form `<appname>/files/<appname>/*` but the contents of `~/.local/share/cinnamon/applets` uses `<appname>/*`. So I don't think it's possible to just clone the repo into `~/.local/share/cinnamon/applets`

What to people do besides lots of moving files around between commits?

**Comment by klangman** [comment]
I created a `git` directory in my home where I cloned the `cinnamon-spaces-applets` repository (along with many others), then I use soft links to link the cloned repository into the `~/.local/share/cinnamon/applets` directory for each applet I am interesting. 
i.e.

```
ln -s ~/git/cinnamon-spices-applets/<applet_name>/files/<applet_name>/ ~/.local/share/cinnamon/applets/<applet_name>
```

**Comment by rcalixte** [comment]
We can make this easier. There is a `test-spice` script that was added to the Actions repository of Spices, we can extend it to the others.

**Comment by rcalixte** [comment, marked as answer]
The `test-spice` script from Actions has been ported for Applets, Desklets, and Extensions and the respective READMEs have been updated to reflect the proper usage.

* [Applets](https://github.com/linuxmint/cinnamon-spices-applets/blob/694c4a0/README.md?plain=1#L61-L83)
* [Desklets](https://github.com/linuxmint/cinnamon-spices-desklets/blob/d7991c7/README.md?plain=1#L61-L83)
* [Extensions](https://github.com/linuxmint/cinnamon-spices-extensions/blob/19e526b/README.md?plain=1#L61-L83)

**Comment by Starcross** [comment]
Really helpful, thanks for this @klangman and @rcalixte

**Comment by OnlineLearningTutorials** [comment]
**Suggesting to add screenshot of applet, desklet, themes in future version of mint**
As of now the desklet chooser of mint is good. But as the applet and desklet increases, it would be better to show directly the screenshot of that desklet, The screenshot would be optional if no screenshot than it will show default image or we can make one main screenshot compulsory, and we can limit the maximum screenshot to be 4 or 8. And the presentation of Desklet chooser be like [cinnamon spices website](https://cinnamon-spices.linuxmint.com/themes/popular) rather than one row per desklet. Also we can the main screenshot to be like 600px by 400px or be any size, we can change icon to main screenshot or to be remain both. The interface of cinnamon spices website is very good we want something like that in over local desklet manager. I am talking of future version, current system is also good.
![desklet manager](https://github.com/user-attachments/assets/e99cdec4-8eb1-445b-9817-a05925dc89da)
![desklet manager future layout](https://github.com/user-attachments/assets/3fb80a82-689a-46da-bf50-8e7677bb737a)

Also directory structure I can sugget `<appname>/<files of app>+<info folder/info folders file that describe metadata etc, can store screenshots here >`
However that depends on mint team what they prefer, because if my directory structure will be followed than we can copy all applet by cloning the git repo however it have some extra infomation in info folder.
